src/Sarif/ConsoleLogger.cs

```diff
@@ -94,40 +94,47 @@ public void Log(IRule rule, Result result)
 
             WriteToConsole(
                 result.Level,
+                result.Kind,
                 physicalLocation?.FileLocation?.Uri,
                 physicalLocation?.Region,
                 result.RuleId,
                 message);
         }
 
-        private void WriteToConsole(ResultLevel level, Uri uri, Region region, string ruleId, string message)
+        private void WriteToConsole(FailureLevel level, ResultKind kind, Uri uri, Region region, string ruleId, string message)
         {
+            // If we are not in an actual failure case, the message level
+            // is not relevant so we'll explicitly set it to None.
+            if (kind != ResultKind.Fail)
```

Comment:
>kind [](start = 16, length = 4)

We could insist that callers maintain this invariant instead of correcting the data ourselves. ?

Reply:
Yes, I think so. If `kind` is not `fail`, but `level` is (say) `error`, then how do we know what's right? Did they mean to set `kind` to `fail`? or did they mean to set `level` to `none`? I'd throw `ArgumentException` if they're inconsistent.

---
In reply to: [252549059](https://github.com/Microsoft/sarif-sdk/pull/1249#discussion_r252549059) [](ancestors = 252549059)

Reply:
agree

---
In reply to: [252892432](https://github.com/Microsoft/sarif-sdk/pull/1249#discussion_r252892432) [](ancestors = 252892432,252549059)